src/Authentication.Abstractions/AzureSession.cs

```diff
@@ -260,14 +263,10 @@ public void UnregisterComponent<T>(string componentName) where T : class
                 () =>
                 {
                     var key = new ComponentKey(componentName, typeof(T));
-                    if (_componentRegistry.ContainsKey(key))
+                    var components = _componentRegistry as ConcurrentDictionary<ComponentKey, object>;
```

Comment:
Can we rename the type from IDictionary to ConcurrentDictionary to reduce the implicit cast?

Reply:
1. AzureSession.Instance is a singleton shared across threads, so it's intended that all the components are also singletons. It's OK if a component is registered multiple times, because all threads will use the same component after all.
2. Sure. I've updated that.